backend/sns/notification/repository.py

```diff
@@ -0,0 +1,135 @@
+from sqlalchemy.orm import Session
+
+from sns.notification.model import Notification
+
+
+class NotificationDB:
+    def get_notification_by_id(
+        self,
+        db: Session,
+        notification_id: int,
+    ) -> Notification:
+        """notification_id에 해당하는 Notification 객체를 조회한다.
+
+        Args:
+            db (Session): db session
+            notification_id (int): 조회할 Notification의 follow_id 속성 값
+
+        Returns:
+            Notification: 조회된 Notification 객체. 해당되는 객체가 없으면 None을 반환
+        """
+        return db.query(Notification).filter(Notification.id == notification_id).first()
```

Comment:
primary key 로 조회를 할 때는 [Query.get()](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.get) 을 사용하는 것이 좋습니다. 

https://stackoverflow.com/questions/6750017/how-to-query-database-by-id-using-sqlalchemy

Reply:
@JeHa00  
달아주신 sqlalchemy 공식문서 링크를 클릭해서 들어가보니 

> Deprecated since version 1.4: The [Query.get()](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.get) method is considered legacy as of the 1.x series of SQLAlchemy and becomes a legacy construct in 2.0. The method is now available as [Session.get()](https://docs.sqlalchemy.org/en/14/orm/session_api.html#sqlalchemy.orm.Session.get) (Background on SQLAlchemy 2.0 at: [Migrating to SQLAlchemy 2.0](https://docs.sqlalchemy.org/en/14/changelog/migration_20.html))

위와 같이 적혀있어서 `Session.get` 을 사용해보도록 하겠습니다!

Reply:
session 은 사용 방식이 아예 다릅니다. session 방식 적용하려면 지금까지 작성한 crud 코드를 다 뜯어 고쳐야 합니다. 
저희 회사에서는 query 를 사용하다 session 방식을 도입했는데, 이는 async 적용을 위해 필수적이었기 때문입니다. 실제 사용은 query 가 더 쉽습니다. 기능적인 차이는 거의 없습니다.